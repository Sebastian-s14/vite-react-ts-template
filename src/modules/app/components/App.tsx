import { useState } from 'react'
// import logo from '../svg/logo.svg'
// import './App.css'

export const App = () => {
  const [count, setCount] = useState(0)

  return (
    <div className="text-center">
      <header className="flex flex-col gap-6 justify-center items-center h-screen text-xl text-white bg-[#282c34]">
        {/* <img src={logo} className="App-logo" alt="logo" /> */}
        <p>Hello Vite + React!</p>
        <p>
          <button
            aria-label="Incrementar"
            className="py-2 px-4 text-black bg-white"
            type="button"
            onClick={() => setCount((count) => count + 1)}
          >
            count is: {count}
          </button>
        </p>
        <p>
          Edit <code>App.tsx</code> and save to test HMR updates.
        </p>
        <p>
          <a
            className="text-[#61dafb]"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          {' | '}
          <a
            className="text-[#61dafb]"
            href="https://vitejs.dev/guide/features.html"
            target="_blank"
            rel="noopener noreferrer"
          >
            Vite Docs
          </a>
        </p>
      </header>
    </div>
  )
}
