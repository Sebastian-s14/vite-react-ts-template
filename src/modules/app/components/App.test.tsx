import { fireEvent, render, screen } from '@testing-library/react'
import { App } from './App'

describe('Pruebas en App', () => {
  test('Debe de retornar un elemento', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
  test('Debe retonar el texto Hello Vite + React!', () => {
    render(<App />)
    expect(screen.getByText('Hello Vite + React!')).toBeTruthy()
  })
  test('Debe retonar valor inicial 0', () => {
    render(<App />)
    // screen.getByRole('button', { name: 'Incrementar' })
    expect(screen.getByRole('button', { name: 'Incrementar' })).toBeTruthy()
  })
  test('Debe aumentar en una unidad al hacer clic en el bonton', () => {
    render(<App />)
    fireEvent.click(screen.getByRole('button', { name: 'Incrementar' }))
    expect(screen.getByText('count is: 1')).toBeTruthy()
  })
})
