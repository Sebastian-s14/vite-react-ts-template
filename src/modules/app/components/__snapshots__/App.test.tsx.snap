// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en App Debe de retornar un elemento 1`] = `
<div>
  <div
    class="text-center"
  >
    <header
      class="flex flex-col gap-6 justify-center items-center h-screen text-xl text-white bg-[#282c34]"
    >
      <p>
        Hello Vite + React!
      </p>
      <p>
        <button
          aria-label="Incrementar"
          class="py-2 px-4 text-black bg-white"
          type="button"
        >
          count is: 
          0
        </button>
      </p>
      <p>
        Edit 
        <code>
          App.tsx
        </code>
         and save to test HMR updates.
      </p>
      <p>
        <a
          class="text-[#61dafb]"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
         | 
        <a
          class="text-[#61dafb]"
          href="https://vitejs.dev/guide/features.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Vite Docs
        </a>
      </p>
    </header>
  </div>
</div>
`;
